lib/datadog/core/configuration/config_helper.rb

```diff
@@ -0,0 +1,108 @@
+          # configuration. Otherwise, it raises an error.
+          #
+          # @param name [String] Environment variable name
+          # @param default_value [String, nil] Default value to return if the environment variable is not set
+          # @param env_vars [Hash] Environment variables to use (ENV, fleet config file hash or local config file hash)
+          # @param source [String] Source of the environment variables (can be 'environment' or 'local/fleet stable config)
+          # @return [String, nil] The environment variable value
+          # @raise [RuntimeError] if the configuration is not supported
+          def get_environment_variable(name, default_value = nil, env_vars: ENV)
+            # datadog-ci-rb is using dd-trace-rb config DSL, which uses this method.
+            # Until we've correctly implemented support for datadog-ci-rb, we disable config inversion if ci is enabled.
+            if !defined?(::Datadog::CI) &&
+                (name.start_with?('DD_', 'OTEL_') || ALIAS_TO_CANONICAL[name]) &&
+                !SUPPORTED_CONFIGURATIONS[name]
+              if defined?(@raise_on_unknown_env_var) && @raise_on_unknown_env_var # Only enabled for tests!
+                if ALIAS_TO_CANONICAL[name]
+                  raise "Please use #{ALIAS_TO_CANONICAL[name]} instead of #{name}. See docs/AccessEnvironmentVariables.md for details."
+                else
+                  raise "Missing #{name} env/configuration in \"supported-configurations.json\" file. See docs/AccessEnvironmentVariables.md for details."
+                end
+              end
+              # TODO: Send telemetry to know if we ever miss an env var
+              return nil
+            end
+
+            config = env_vars[name]
+            if config.nil? && ALIASES[name]
+              ALIASES[name].each do |alias_name|
+                return env_vars[alias_name] if env_vars[alias_name]
+              end
+            end
+
+            config || default_value
+          end
+
+          def log_deprecated_environment_variables(logger, env_vars: ENV, source: 'environment')
+            @log_deprecations_called_with ||= {}
+            return if @log_deprecations_called_with[source]
+
+            @log_deprecations_called_with[source] = true
+            DEPRECATIONS.each do |deprecated_env_var, message|
+              next unless env_vars.key?(deprecated_env_var)
+
+              # As we only use warn level, we can use a new logger.
+              # Using logger_without_configuration is not possible as it uses an environment variable.
+              Datadog::Core.log_deprecation(disallowed_next_major: false, logger: logger) do
+                "#{deprecated_env_var} #{source} variable is deprecated" +
+                  (ALIAS_TO_CANONICAL[deprecated_env_var] ? ", use #{ALIAS_TO_CANONICAL[deprecated_env_var]} instead." : ". #{message}.")
+              end
+            end
+          end
```

Comment:
Minor: This could be private + `@log_deprecations_called_with` can be simplified to a simple boolean if the logging was made or not (since it's only used by `log_deprecations_from_all_sources`)